packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx

```diff
@@ -299,19 +299,26 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
     }
   };
 
-  private _onGotoToday = (): void => {
+  private _onGotoToday = (selectDateOnClick: boolean): void => {
     const { dateRangeType, firstDayOfWeek, today, workWeekDays } = this.props;
 
     const dates = getDateRangeArray(today!, dateRangeType!, firstDayOfWeek!, workWeekDays!);
 
-    this._onSelectDate(today!, dates);
+    if (selectDateOnClick) {
+      this._onSelectDate(today!, dates);
```

Comment:
I realise there is a lot of preexisting usages of "!" in line 305, but could we try to minimize it (such that we comment all usages of it providing a justification)?

Why are we certain that `today` will never be undefined, if it is presumably an optional prop?